app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/onboarding/SubscriptionController.scala

```diff
@@ -69,6 +69,27 @@ class SubscriptionController @Inject() (
     authenticatedActionWithSubscriptionReady.async { implicit request =>
       val details = request.subscriptionReady.subscriptionDetails
       val result  = for {
+        _                    <- EitherT(
+                                  updateSession(sessionStore, request)(
+                                    _.copy(journeyStatus =
+                                      Some(
+                                        Subscribing(
+                                          SubscriptionDetails(
+                                            name = details.name,
+                                            emailAddress = details.emailAddress,
+                                            address = details.address,
+                                            contactName = details.contactName,
+                                            sapNumber = details.sapNumber,
+                                            emailSource = details.emailSource,
+                                            addressSource = details.addressSource,
+                                            contactNameSource = details.contactNameSource
+                                          ),
+                                          request.subscriptionReady.ggCredId
+                                        )
+                                      )
+                                    )
+                                  )
+                                )
         subscriptionResponse <-
```

Comment:
🚫 This should be surrounded by a lock and there should be a check for the registration status.

Reply:
Not sure why we want to have lock on a subscribe journey twice. We are already using lock on a subscribe call at the backend.

Reply:
Rephrased.